akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCartRoutes.scala

```diff
@@ -0,0 +1,100 @@
+package sample.cqrs
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.cluster.sharding.typed.scaladsl.ClusterSharding
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Route
+import akka.util.Timeout
+
+object ShoppingCartRoutes {
+  final case class AddItem(cartId: String, itemId: String, quantity: Int)
+  final case class UpdateItem(cartId: String, itemId: String, quantity: Int)
+  final case class Checkout(cartId: String)
+}
+
+class ShoppingCartRoutes()(implicit system: ActorSystem[_]) {
+
+  implicit private val timeout: Timeout = 5.seconds
+  private val sharding = ClusterSharding(system)
+
+  import ShoppingCartRoutes._
+  import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+  import akka.http.scaladsl.server.Directives._
+  import JsonFormats._
+
+  val shopping: Route =
+    pathPrefix("shopping-carts") {
+      concat(
+        post {
+          entity(as[AddItem]) {
+            data =>
+              val entityRef = sharding.entityRefFor(ShoppingCart.EntityKey, data.cartId)
+              val reply: Future[ShoppingCart.Confirmation] =
+                entityRef.ask(ShoppingCart.AddItem(data.itemId, data.quantity, _))
+              onSuccess(reply) {
+                case ShoppingCart.Accepted(_) =>
+                  complete(
+                    StatusCodes.Accepted -> s"added item ${data.itemId}: ${data.quantity} to cart ${data.cartId}")
```

Comment:
A bit unconventional accepting Json but responding with text, ok for a sample I guess.

Reply:
changed to return the summary